header/core/listener.go

```diff
@@ -20,24 +21,30 @@ import (
 // broadcasts the new `ExtendedHeader` to the header-sub gossipsub
 // network.
 type Listener struct {
-	bcast     header.Broadcaster
-	fetcher   *core.BlockFetcher
-	bServ     blockservice.BlockService
+	fetcher *core.BlockFetcher
+	bServ   blockservice.BlockService
+
 	construct header.ConstructFn
-	cancel    context.CancelFunc
+
+	broadcastExtendedHeader header.Broadcaster
+	broadcastEDSHash        shrexpush.BroadcastEDSHash
```

Comment:
It's OK to depend on shrexsub here directly. This will make a sub pkg of the `header` depend on `eds`, but it is okay for now unless it does not produce any cycles. After the header interface PR is merged, we will move Listener to the `core` pkg.

Reply:
So pls just make it directly depend

Reply:
Does not it break the encapsulation rule? We do not need the whole service but only one method. So, here I agree with @renaynay about having only one method that is expected to use.

Reply:
It's about abstracting it first of all. I am fine to abstract it though, just do not see a big reason, but let's keep it, ok. The only nit is to call it `shrexpush.BroadcastFn` or `shrexpush.Broadcaster`. The EDSHash suffix is implied by shrexpush as it's only pushes data hashes

Reply:
I agree with @Wondertan then re naming but the reason for why it's abstracted is bc the header.Broadcaster is also "abstracted" - ideally they'd be abstracted in the same way at some point but 🤷🏻

Reply:
renamed